packages/access-client/src/agent.js

```diff
@@ -149,27 +149,38 @@ export class Agent {
    *
    * @param {import('@ucanto/interface').Capability[]} [caps]
    */
-  async *#delegations(caps) {
+  #delegations(caps) {
     const _caps = new Set(caps)
+    /** @type {Array<{ delegation: Ucanto.Delegation, meta: import('./types').DelegationMeta }>} */
+    const values = []
     for (const [, value] of this.#data.delegations) {
       // check expiration
-      if (!isExpired(value.delegation)) {
-        // check if delegation can be used
-        if (!isTooEarly(value.delegation)) {
-          // check if we need to filter for caps
-          if (Array.isArray(caps) && caps.length > 0) {
-            for (const cap of _caps) {
-              if (canDelegateCapability(value.delegation, cap)) {
-                _caps.delete(cap)
-                yield value
-              }
+      if (
+        !isExpired(value.delegation) && // check if delegation can be used
+        !isTooEarly(value.delegation)
+      ) {
+        // check if we need to filter for caps
+        if (Array.isArray(caps) && caps.length > 0) {
+          for (const cap of _caps) {
+            if (canDelegateCapability(value.delegation, cap)) {
+              _caps.delete(cap)
+              values.push(value)
             }
-          } else {
-            yield value
           }
+        } else {
+          values.push(value)
         }
-      } else {
-        // delete any expired delegation
+      }
+    }
+    return values
+  }
+
+  /**
+   * Clean up any expired delegations.
+   */
+  async removeExpiredDelegations() {
```

Comment:
can we call this methods somewhere ? like `fromStore` or any other place thats already async ?

Reply:
I've added to `addProof` and `delegate` for your approval. Makes sense to me if we're mutating delegations then we can remove some old ones.